### PR TITLE
do-syntax

### PR DESCRIPTION
**Pull request by L-as**
```haskell
import qualified Plutarch.Monadic as P

P.do
  PTxInfo y <- pmatch (x :: Term s PTxInfo)
  PTrue <- pmatch $ pcon PFalse -- calls `P.fail errormsg` if there's not a match, which is `ptrace errormsg perror`.
  z <- plet $ x + y
  ptrace "ahduawhdw"
  perror
```

### CODE REVIEW

**Comment by L-as** [comment]
I like this syntax.

File: Plutarch/Monadic.hs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import Plutarch.Trace (ptrace)
+
+(>>=) :: a -> a
+(>>=) = id
+
+(>>) :: a -> a
+(>>) = id
+
+fail :: String -> Term s a
+fail msg = ptrace (fromString msg) perror
```
Comment (L-as) [inline Plutarch/Monadic.hs:16]: ptraceError

**Review by L-as — COMMENTED** [review]
I need to change this.

**Review by t1lde — COMMENTED** [review]
Is this back-compatible with ghc<9? I don't think we want to force users into ghc9 (yet)

**Comment by L-as** [comment]
I've tried not to do anything that will cause the base library to fail to compile with GHC 8.10, albeit the tests won't work obviously.

**Comment by t1lde** [comment]
should be fine, plutus itself is probably likely to start requiring ghc9 anyway

**Review by t1lde — APPROVED** [review]

**Pull request merged**